Sources/Sentry/SentryTracer.m

```diff
@@ -462,6 +470,9 @@ - (void)finishInternal
             }
             profiler = nil;
         }
+#    if SENTRY_HAS_UIKIT
+        [SentryFramesTracker.sharedInstance resetProfilingTimestamps];
```

Comment:
`h`: I think this could lead to a race condition. Although we currently only create auto-generated transactions on the main thread, this could become a problem. I think we should get a local copy of the timestamps on L:416 and also call `SentryFramesTracker.sharedInstance resetProfilingTimestamps]` after L:416. The same applies to the `profiler`. I think we should as well get a local copy in the lock around L:416 and already set the class property to nil, but we can do this in a different PR.

Reply:
I see what you're saying here. There are a few things that can be cleaned up here.